apps/src/code-studio/clientState.js

```diff
@@ -91,12 +93,16 @@ clientState.writeSourceForLevel = function(
 
 /**
  * Tracks the lines of code written after the user clicks run if their
- * solution is successful.
+ * solution is successful. Skips if the user is logged in.
  * @param {boolean} result - Whether the user's solution is successful
  * @param {number} lines - Number of lines of code user wrote in this solution
  */
 clientState.trackLines = function(result, lines) {
-  if (result && isFinite(lines)) {
+  if (
+    result &&
+    isFinite(lines) &&
+    !getStore().getState().progress.usingDbProgress
```

Comment:
Just checking -- can .progress be undefined here?

Reply:
Had to double check that one. It won't be undefined. `progress` will be registered when the redux module is imported.